src/plugins/appeals/unbanThread.ts

```diff
@@ -0,0 +1,35 @@
+import {useEvent} from "../../hooks";
+import {Interaction, Message} from "discord.js";
+
+useEvent("interactionCreate", async (interaction: Interaction) => {
+    if (!interaction.isButton()) {
+        return;
+    }
+    const {customId} = interaction;
+    const [interactionType, action, userTag, userId] = customId.split("-");
+    const {message} = interaction;
+    if (interactionType !== "appeal") {
+        return;
+    }
+    if (action !== "thread") {
+        return;
+    }
+    try {
+        await interaction.message.startThread ({
+            name: `Appeal Thread - ${userTag}`
+        })
+    } catch {
+        await interaction.reply({
```

Comment:
G'day's permissions are limited in the staff server, as such we cannot assume that `startThread` failing means a thread already exists. Instead we need to catch the error and use its error code (`code` property included in `DiscordjsError`s) to determine why there was an error, and inform the user accordingly.

Reply:
Added a catch for error “MessageExistingThread”. If true, it runs a silent reply stating, “A thread was already created.” Otherwise, it’ll throw a generic error but write to the console.